pytorch/pytorch_edgeml/graph/rnn.py

```diff
@@ -1117,12 +1137,25 @@ def forward(self, x, brickSize):
         x_bricks = torch.reshape(x_bricks, [oldShape[0], oldShape[1] * oldShape[2], oldShape[3]])
         # x bricks: [brickSize, numBricks * batchSize, featureDim]
         # x_bricks = torch.Tensor(x_bricks)
-        hidd0, out0 = self.rnn0(x_bricks)
+        dropoutLayer_0 = nn.Dropout(p = self.dropoutProbability_l0)
```

Comment:
Nope.

Lets change the way you are dealing with dropout to make it crisp and short.


```
# at the top of __init__, just assign this to self. variables
self.dropoutProbability_l0 = dropoutProbability_l0
self.dropoutProbability_l1 = dropoutProbability_l1
# . . .
# rest of the code till you define the cells
# . . .

self.dropoutLayer0 = None
self.dropoutLayer1 = None
hidd0 = rnn0(whatever)
if self.dropoutProbability0 is not None:
    self.dropoutLayer0 = nn.Dropout() # define this
    hidd0 = self.dropoutLayer0(hidd0, . . .)

hidd1 = self.rnn1(hidd0, . . .)
if self.dropoutProbability1 is not None:
    self.dropoutLayer1 = nnDropout()
    hidd1 = self.dropout(. . .)
# . . .
# Continue to FC layer
# . . .
    
```